Hippo.Web/ClientApp/src/components/Order/OrderForm.tsx

```diff
@@ -0,0 +1,344 @@
+import React, { useState } from "react";
+import { FormProvider, useForm, useWatch } from "react-hook-form";
+import { OrderModel } from "../../types";
+import { Form } from "reactstrap";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faDollarSign } from "@fortawesome/free-solid-svg-icons";
+import FormSubmitButton from "../../Shared/Form/FormSubmitButton";
+import MetaDataFields from "./MetaDataFields";
+import OrderFormField from "./OrderFormField";
+import OrderFormTotalFields from "./OrderFormTotalFields";
+import { authenticatedFetch } from "../../util/api";
+import BillingsFields from "./BillingsFields";
+import { ShowFor } from "../../Shared/ShowFor";
+
+interface OrderFormProps {
+  orderProp: OrderModel;
+  readOnly: boolean;
+  isAdmin: boolean;
+  cluster: string;
+  onlyChartStrings: boolean;
+  onSubmit: (order: OrderModel) => Promise<void>;
+}
+
+const OrderForm: React.FC<OrderFormProps> = ({
+  orderProp,
+  readOnly,
+  isAdmin,
+  cluster,
+  onlyChartStrings,
+  onSubmit,
+}) => {
+  const methods = useForm<OrderModel>({
+    defaultValues: orderProp,
+    mode: "onBlur",
+  });
+  const {
+    handleSubmit,
+    setError,
+    formState: { isDirty, isSubmitting },
+  } = methods;
+
+  const [foundPI, setFoundPI] = useState(null);
+  const [localInstallmentType, setLocalInstallmentType] = useState(
+    methods.getValues("installmentType"),
+  );
+  const [limitedEditing, setLimitedEditing] = useState(false);
+
+  React.useEffect(() => {
+    const newStatus = orderProp.status;
+    methods.setValue("status", newStatus);
```

Comment:
i think it'd be nicer to extract the `setValue` and `getValues` methods from `method` instead of using `methods.setValue` and then needing to useeffect depend on the entire method object.  Also, why call it methods, maybe at least formMethods but that doesn't really work either because formState is not a method.

Reply:
```jsx
  const {
    handleSubmit,
    setError,
    setValue,
    getValues,
    formState: { isDirty, isSubmitting },
  } = methods;

  const [foundPI, setFoundPI] = useState(null);
  const [localInstallmentType, setLocalInstallmentType] = useState(
    getValues("installmentType"),
  );
  const [limitedEditing, setLimitedEditing] = useState(false);

  React.useEffect(() => {
    const newStatus = orderProp.status;
    setValue("status", newStatus);

    if (
      newStatus === "Active" ||
      newStatus === "Rejected" ||
      newStatus === "Cancelled" ||
      newStatus === "Completed"
    ) {
      setLimitedEditing(true);
    } else {
      setLimitedEditing(false);
    }

    const newInstallmentDate = orderProp.installmentDate;
    const newExpirationDate = orderProp.expirationDate;
    setValue("installmentDate", newInstallmentDate);
    setValue("expirationDate", newExpirationDate);
  }, [
    setValue,
    orderProp.expirationDate,
    orderProp.installmentDate,
    orderProp.status,
  ]);
```

the main difference here is that you have `setValue` instead of `methods` in your effect dependency array. this means that instead of the effect triggering every time the `methods` _object_ changes, you're just targeting the `setValue` property. potentially causing a lot less re-runs of the effect. normally, having an entire object in your dependency array will trigger the effect on every re-render, because of the way javascript handles object equality. 

```jsx
const MyComponent = () => {
  const [state, setState] = React.useState({ key: 'value' });

  React.useEffect(() => {
    console.log('Effect ran');
  }, [state]);

  return (
    <div>
      <button onClick={() => setState({ key: 'value' })}>Update State</button>
    </div>
  );
};
```

the effect here will run every time the button is clicked, even though the value hasn't actually changed, because js using reference equality for objects, arrays, and functions. like, 

```js
const obj1 = { a: 1 };
const obj2 = { a: 1 };
const obj3 = obj1;

console.log(obj1 === obj2); // false
console.log(obj1 === obj3); // true
```

so the object reference can change quite a lot on re-renders, even if the actual contents are the same. this is where `useMemo()` is useful, to say "this is the same object unless the dependencies change", and `useCallback()` similarly for functions.

so instead:
```jsx
const MyComponent = () => {
  const [state, setState] = React.useState({ key: 'value' });

  React.useEffect(() => {
    console.log('Effect ran');
  }, [state.key]);

  return (
    <div>
      <button onClick={() => setState({ key: 'value' })}>Update State</button>
    </div>
  );
};
```
here, targeting the `key` only, the effect will only run when the _string_ 'value' changes, since in js `'value' === 'value'` returns true where `obj1 = { key: 'value'} === obj2 = {key: 'value'}` returns false. 

tl;dr it's safer to be as explicit in your deps as possible. possibly `react-hook-form` has thought of this and the `methods` object is memoized, but ya never know. also, javascript is weird

Reply:
#394